devdoc/sm_requirements.md

```diff
@@ -198,6 +201,23 @@ MOCKABLE_FUNCTION(, int, sm_close_begin, SM_HANDLE, sm);
 
 **SRS_SM_02_071: [** If there are any failures then `sm_close_begin` shall fail and return `SM_ERROR`. **]**
 
+### sm_close_begin_with_cb
+```c
+MOCKABLE_FUNCTION(, SM_RESULT, sm_close_begin_with_cb, SM_HANDLE, sm, SM_CLOSING_COMPLETE_CALLBACK, callback, void*, callback_context);
+```
+
+`sm_close_begin_with_cb` asks from `sm` permission to exit `SM_OPENED` state (or one of its derived state) and return to `SM_CREATED`. `sm_close_begin_with_cb` invokes the `callback` function with `callback_context` before waiting for pending calls to become 0.
+
+**SRS_SM_28_001: [** If `sm` is `NULL` then `sm_close_begin_with_cb` shall fail and return `SM_ERROR`. **]**
+
+**SRS_SM_28_002: [** If `callback` is `NULL` then `sm_close_begin_with_cb` shall fail and return `SM_ERROR`. **]**
+
+**SRS_SM_28_003: [** `sm_close_begin_with_cb` shall behave as if `sm_close_begin` was called except as follows.  **]**
```

Comment:
>SRS_SM_28_003 [](http://example.com/codeflow?start=2&length=13)

Now I'm curious, how did traceability pass? You have new specs (SRS and not something like S_R_S that's in code) but no new tests. Is our traceability test broken?

Reply:
Nevermind, codeflow was just showing me some old iteration I think